src/routes/blog/spyder-6-remote-development/+page.md

```diff
@@ -0,0 +1,94 @@
+
+That project is managed through an internal Spyder plugin, *Remote Client*, which uses the Jupyter API to manage remote Spyder kernels and connect them to Spyder.
+The *Remote Client* [frontend](https://github.com/spyder-ide/spyder/pull/22079) and [backend](https://github.com/spyder-ide/spyder/pull/21757) was developed as an interface for Spyder to connect and manage the Jupyter Server installed on the remote machine.
+The plugin connects to the machine over SSH, installs the server, and creates SSH tunnels for the exposed server and extra APIs implemented by `spyder-remote-services`.
+The remote client plugin is currently able to automatically create remote [IPython Consoles](https://docs.spyder-ide.org/current/panes/ipythonconsole.html) on behalf of users.
+Furthermore, you can now stop remote computations and restart remote kernels, which was not possible before.
+
+
+![Spyder consoles menu listing remote as well as local Python environments to open](remote-console-menu.png)
+
+## What challenges did you face and how'd you overcome them?
+
+There are a number of Python libraries that implement the SSH protocol, but most of them use the underlying SSH client and server installed by the operating system.
+Unfortunately, the Windows SSH client has a serious issue that prevents tunnels from working correctly, so we needed an implementation that uses the underlying SSH libraries directly.
+After investigating a number of options, the [AsyncSSH project](https://github.com/ronf/asyncssh) was the only one that both meets that requirement and is also robust and well maintained, which we implemented in the PR [spyder-ide/spyder#22223](https://github.com/spyder-ide/spyder/pull/22223).
+
+However, there was still an important problem with AsyncSSH: Spyder uses the Qt framework's event loop for its graphical user interface, which is not async-enabled.
+Therefore, we had to write an async API that was flexible enough to be able to call several async functions in a specific event loop, allow for loops to be run concurrently (to avoid blocking the main Qt event loop and causing Spyder's GUI to freeze), and be thread-safe so it can be called from any Qt thread.
+
+To cover all those requirements, Hendrick created the [`@AsyncDispatcher` decorator](https://github.com/spyder-ide/spyder/blob/v6.0.0/spyder/api/asyncdispatcher.py#L39) function, which starts a thread to run an async loop, schedules the async function in a specific loop, and returns a future object with the function's result.
+That result is emitted in a Qt signal which is then used by Spyder synchronously to perform other tasks (e.g. check if the server is running).
+
+
+![Spyder remote connection manager, showing a successfully connected host](remote-connection-manager-status.png)
+
+## What have you learned so far?
+
+Working on an open source project has provided me with the opportunity to witness how my contributions directly impact users in real time.
```

Comment:
Same issue as before, adding the name again here isn't appropriate since it is a direct question in a heading directed at the already named and identified interview subject rather than a formal back and forth with both parties having their name identified this way